src/client/Microsoft.Identity.Client/Internal/Requests/OnBehalfOfRequest.cs

```diff
@@ -191,7 +191,8 @@ private async Task<AuthenticationResult> FetchNewAccessTokenAsync(CancellationTo
             }
 
             if (msalTokenResponse.ClientInfo is null &&
-                AuthenticationRequestParameters.AuthorityInfo.AuthorityType != AuthorityType.Adfs)
+                AuthenticationRequestParameters.AuthorityInfo.AuthorityType != AuthorityType.Adfs &&
+                AuthenticationRequestParameters.AuthorityInfo.AuthorityType != AuthorityType.Dsts)

```

Comment:
Again, we should have a flag on Authority that encapsulates this logic, maybe "SupportsServicePrincipalObo". Also, DSTS is more like AAD, not like ADFS.